src/code/sys_cmpdma.c

```diff
@@ -19,14 +19,152 @@
 
 #pragma GLOBAL_ASM("asm/non_matchings/code/sys_cmpdma/func_80178A94.s")
 
+typedef union {
+    struct {
+        u8 r;
+        u8 g;
+        u8 b;
+        u8 a;
+    };
+    s32 rgba;
+} Struct1_80178AC0;
+
+typedef union {
+    u16 a, b;
+    s32 ab;
+} Struct2_80178AC0;
+
+typedef struct {
+    /* 0x00 */ u32 data1;
+    /* 0x00 */ u32 data2;
+} CmpDmaBssStruct;
+
+// static CmpDmaBssStruct D_801FBBE0;
+extern CmpDmaBssStruct D_801FBBE0;
+
+#ifdef NON_MATCHING
+// Some constants re-ordered at the top
```

Comment:
Forgot in my other review, but would you be able to post a scratch for this? Unless you still want to work on it.

Reply:
https://decomp.me/scratch/OSthZ